packages/server/src/scripts/import_arpa_reporter_dump.js

```diff
@@ -0,0 +1,469 @@
+        .mapValues("main_agency_id")
+        .value();
+    const usersToCreate = dbContents.users.map((user) =>
+        rekeyForeignKeys(
+            {
+                email: user.email,
+                name: user.name,
+                // Note: ARPA Reporter had a "reporter" role. For our purposes,
+                // anything non-admin becomes "staff"
+                role_id: user.role === 'admin' ? adminRole : staffRole,
+                tenant_id: user.tenant_id,
+                // Copied users belong to the main agency of their tenant
+                agency_id:
+                    mainAgencyByTenant[idLookupByTable.tenants[user.tenant_id]],
+            },
+            idLookupByTable,
+            ["agency_id"]
+        )
+    );
+
+    // Check if there are any existing users with the specified emails
+    const usersWithDupeEmails = await knex("users")
+        .select("email")
+        .whereIn("email", _.map(usersToCreate, "email"));
+    if (usersWithDupeEmails.length != 0) {
+        console.error(
+            "Found duplicate emails",
+            _.map(usersWithDupeEmails, "email")
+        );
+        throw new Error(
+            "found users with duplicate emails! Delete them first, or update script to handle."
+        );
+    }
+
+    // Do the inserts
+    const inserted = await knex("users").insert(usersToCreate).returning("*");
+    const idLookup = _.chain(inserted)
+        .map((insertedRow, idx) => [dbContents.users[idx].id, insertedRow.id])
+        .fromPairs()
+        .value();
+
+    return { inserted, idLookup };
+}
+
+const specialTableHandlers = {
+    tenants: importTenants,
+    users: importUsers,
+    agencies: importAgencies,
+};
+
+async function importDatabase(dbContents) {
+    const idLookupByTable = {};
+    const insertedRowsByTable = {};
+
+    for (const tableName of TABLES) {
+        console.log("Importing table", tableName, "...");
+
+        const specialHandler = specialTableHandlers[tableName];
+        const { inserted, idLookup } = specialHandler
+            ? await specialHandler(
+                  dbContents,
+                  idLookupByTable,
+                  insertedRowsByTable
+              )
+            : await importTable(
+                  tableName,
+                  dbContents[tableName],
+                  idLookupByTable,
+                  insertedRowsByTable
+              );
+
+        idLookupByTable[tableName] = {
+            ...idLookupByTable[tableName],
+            ...idLookup,
+        };
+        insertedRowsByTable[tableName] = [
+            ...insertedRowsByTable[tableName],
+            ...inserted,
+        ];
+    }
+
+    return { idLookupByTable, insertedRowsByTable };
+}
+
+// based on function of the same name in ARPA's services/persist-upload module
+// with some adjustments. Ideally, we'd just import it, but this migrate script
+// is going to be committed before ARPA's code is copied in.
+function uploadFSName(upload) {
```

Comment:
Since these are being duplicated here, can you write some big caps lock warnings in the ARPA repo PR saying that logic cannot be modified, just to warn people off in case they haven't seen this?

Reply:
Yeah, I'll do that in the ARPA-side PR.

In practice though, these are unlikely to be changed since they are used to reference files already on disk and any substantial change to the logic here would break those references too.